torch/csrc/jit/codegen/cuda/scheduler/utils.h

```diff
@@ -289,21 +290,85 @@ namespace matmul_utils {
 TORCH_CUDA_CU_API void scheduleContiguousVectorLoad(
     TensorView* tv,
     MatMulTileOptions tile,
-    int vector_word);
+    int vector_word,
+    bool vectorize = true);
 
 //! Schedule utility for mma output in matmul main loop:
 //!  Realize the hierarchical tiling based on the given tiling options.
+//! TODO: rewrite this one with makeTile
 TORCH_CUDA_CU_API void scheduleWarpTileWithReduction(
     TensorView* tv,
     MatMulTileOptions tile);
 
 //! Schedule utility for mma output in matmul main loop:
 //!  Realize the hierarchical tiling based on the given tiling options
 //! on consumers of mma ops in epilog.
+//! TODO: remove this one eventually.
 TORCH_CUDA_CU_API void scheduleWarpTileWithNoReduction(
     TensorView* tv,
     MatMulTileOptions tile);
 
+//! Lower level primitive spliting inner iterdomains into tiles:
+//! Eg.
+//!  A[B,I0,I1,I2] -> makeTile({1,2,3})
+//! Gives A[B, I0o, I1o, I2o, I0i(1), I1i(2), I2i(3)]
+TORCH_CUDA_CU_API void makeTile(TensorView* tv, std::vector<int> tile_sizes);
```

Comment:
This function is mainly used on the output of mma ops, so trivial reductions on batch dimension or separately coming from either operand would not show up here. 

I guess the only remaining trivial dimensions here would be trivial K-axis, i.e. the size-1 dimensions that are on both operands and are reduced at mma op. This would require some build out in different scenarios, most obvious case I could think of is when only trivial K-axes exist and this mma should be converted to a pointwise op.

Reply:
Not necessarily a trivial reduction happening at the same exact point but a dimension that maps to a trivial reduction which could cause trouble when trying to inline. As inlining a broadcast dimension that's never broadcasted is trivial, but you can't inline a trivial reduction. Something like this:
https://github.com/csarofeen/pytorch/blob/devel/torch/csrc/jit/codegen/cuda/scheduler/reduction_utils.cpp#L333-L344

Reply:
I see. Mapped trivial dimensions would be tricky. I will try to add a few cases along this line in follow ups.